websites/S/Stremio/presence.ts

```diff
@@ -0,0 +1,179 @@
+const presence = new Presence({
+		clientId: "969208766807547917"
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let video: HTMLMediaElement,
+	timestamp: [number, number],
+	pauseCheck: boolean,
+	search: HTMLInputElement,
+	genreSort: string,
+	title: string;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "logo",
+			startTimestamp: browsingTimestamp
+		},
+		page = window.location.href,
+		[privacy, thumbnails, buttons] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("thumbnails"),
+			presence.getSetting<boolean>("buttons")
+		]),
+		active = document.querySelector(
+			"[class='ng-binding ng-scope selected']"
+		)?.textContent;
+
+	if (window.location.hostname.includes("app.strem.io")) {
+		search = document.querySelector("#global-search-field");
+		video = document.querySelector<HTMLMediaElement>("#videoPlayer");
+		if (privacy && !video) presenceData.details = "Browsing...";
+		else if (privacy && video) presenceData.details = "Watching...";
+		else if (search?.value) {
+			presenceData.details = "Searching For:";
+			presenceData.state = search.value;
+		} else if (page.includes("/detail/")) {
+			title = document.querySelector(
+				"#detail > div:nth-child(3) > div > div.sidebar-info-container > div > div.logo > div"
+			).textContent;
+			presenceData.details = title;
+			presenceData.buttons = [
+				{
+					label: "Watch Video",
+					url: page
+				}
+			];
+			if (thumbnails) {
+				presenceData.largeImageKey =
+					document
+						.querySelector(
+							"#detail > div.details-less-info > div.details-top > div:nth-child(1)"
+						)
+						?.firstElementChild.getAttribute("src") ?? "logo";
+			}
+		} else if (page.includes("addons")) {
+			search = document.querySelector(
+				"#addons > div.filter > form > div.addon-search-input > input"
+			);
+			if (search?.value) {
+				presenceData.details = "Searching Addons For:";
+				presenceData.state = search.value;
+			} else {
+				presenceData.state = active ?? "All";
+				title = document.querySelector(
+					"[class='ng-scope selected']"
+				)?.textContent;
+
+				presenceData.buttons = [
+					{
+						label: "Browse Addons",
+						url: page
+					}
+				];
+				presenceData.details = `Browsing ${title}:`;
+			}
+		} else if (page.includes("settings")) {
+			presenceData.details = `${
+				document.querySelector("[class='ng-scope ng-binding active']")
+					?.textContent ?? "General"
+			} Settings`;
+		} else if (page.includes("/discover/")) {
+			if (active) {
+				genreSort = document.querySelector(
+					"[class='ng-scope ng-binding selected']"
+				)?.textContent;
+				if (!genreSort) {
+					genreSort =
+						document.querySelectorAll(
+							"[class='ng-binding ng-scope selected']"
+						)[2]?.textContent ?? "None";
+				}
+				if (genreSort === "Top") genreSort = "All";
+				presenceData.buttons = [
+					{
+						label: `Browse ${active}`,
```

Comment:
Can you show an example of this?

Reply:
![image](https://user-images.githubusercontent.com/42322979/169229863-165976ba-46f8-47a2-be1b-117cc05fa56f.png)